### PR TITLE
fix(ir): verify pl.split_aiv placement and close silent split skips

### PR DESCRIPTION
**Pull request by Hzfengsy**
## Summary

`pl.split_aiv` opens a CORE_GROUP-level region, so it may only be authored inside a
`pl.at(level=CORE_GROUP)` scope or at the top of an Opaque function — never inside a
`FunctionType.InCore` body. That rule was previously enforced only as a side effect:
`LowerAutoVectorSplit` simply failed to lower the region, twelve passes later. It is now
stated directly by `AivSplitValid` check (h), which fires right after `OutlineIncoreScopes`.

Making the check independent of IR shape also let the parser stop wrapping a top-level
region in a synthesized `InCore` scope, which is what restores `parse(print(p)) == p` after
pass 8. Authors see an earlier, actionable diagnostic for the misplaced form; test authors
get the print→parse roundtrip instrument back for every `pl.split_aiv` program, and pass-8
dumps become reparseable again. Two paths that previously skipped work *silently* now
report instead.

No interface change and no migration step: the authoring form this rejects was already
rejected, only later and with a less direct message.

## Changes

- `src/ir/verifier/verify_aiv_split.cpp`: adds `AivSplitValid` check (h). It keys on
  **provenance** — the `split_aiv` attr `OutlineIncoreScopes` stamps on a function it mints
  from a region-bearing scope — rather than on IR shape, so it still rejects a hand-authored
  region in an `InCore` function now that the parser emits such a region bare. A shape-based
  test would have gone silently inert.
- `python/pypto/language/parser/ast_parser.py`: `_parse_split_aiv_for_loop` decides whether to
  synthesize an `InCore` wrapper from the enclosing function's type, not from an open-scope
  test. Inside a `FunctionType.InCore` body no scope is open, so the old test wrapped the
  region and reparsing an outlined `*_incore_0` produced `InCoreScopeStmt != SplitAivScopeStmt`.
- `src/ir/transforms/lower_auto_vector_split_pass.cpp`: closes two silent skips.
  `RollupAffinity` had no `ScopeStmt` arm, so a scope-bodied AUTO `pl.split` function rolled up
  `SHARED`, read as not-mixed, and was passed through completely unsplit with no diagnostic; it
  now classifies through the scope, and the AUTO arm rejects a mixed function it cannot lower
  while still passing a pure-vector one through unchanged. Separately, the inner region walkers
  stepped over a scope nested *inside* a region body, leaving its vector ops full-width on both
  AIV lanes; the post-lowering guard now rejects any surviving scope, not only a surviving region.
- `tests/ut/…`: adds regression coverage for each guard, including the boundary cases that keep
  them from over-rejecting (an outlined `InCore` function passes; a pure-vector scope-bodied
  `pl.split` is still passed through). Removes three roundtrip-instrument workarounds that the
  print→parse asymmetry had forced, so those paths are back under full roundtrip coverage.
- `python/pypto/language/dsl_api.py`, `docs/{en,zh}/dev/passes/{20-lower_auto_vector_split,99-verifier}.md`:
  document the placement rule, the provenance keying, and both new guards.

## Verification

- `cmake --build build --parallel "$PYPTO_BUILD_JOBS"`: exit 0
- `python -m pytest tests/ut/ -n "$PYPTO_TEST_JOBS" -q`: exit 0 — 9808 passed, 3 skipped
- `python -m pytest` on the six affected test files: exit 0 — 391 passed
- `python -m pytest tests/st/codegen/dsl/ -k split -q`: exit 0 — 13 passed (on device, run before the final rebase)
- `ruff check python/ tests/`: exit 0
- `tests/lint/check_docs_en_zh_parity.py`, `check_english_only.py`, `check_headers.py`, `check_docs_nav.py`: exit 0 each

Each new guard was also checked negatively: with the corresponding fix reverted, its regression
test fails.

### CODE REVIEW

File: tests/ut/ir/transforms/test_convert_tensor_to_tile_ops.py [hunk 1]
```diff
@@ -5473,28 +5508,28 @@ def main_incore_0(
...
             ) -> pl.Tensor[[256, 128], pl.FP32]:
+                pl.func_attr({"split_aiv": True, "split": pl.SplitMode.UP_DOWN})
                 x__tile: pl.Tile[[128, 128], pl.FP32, pl.Mem.Vec] = pl.tile.load(
                     x, [0, 0], [128, 128], [128, 128], target_memory=pl.Mem.Vec
                 )
-                with pl.at(level=pl.Level.CORE_GROUP):
-                    for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.UP_DOWN):
-                        h__tile: pl.Tile[[128, 128], pl.FP32, pl.Mem.Vec] = pl.tile.exp(x__tile)
-                        res__tile: pl.Tile[[256, 128], pl.FP32, pl.Mem.Mat] = pl.tile.aic_gather(h__tile)
+                for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.UP_DOWN):
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_convert_tensor_to_tile_ops.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_convert_tensor_to_tile_ops.py [hunk 2]
```diff
@@ -5509,18 +5544,18 @@ def main_incore_0(
...
             ) -> pl.Tensor[[128, 256], pl.FP32]:
+                pl.func_attr({"split_aiv": True, "split": pl.SplitMode.LEFT_RIGHT})
                 x__tile: pl.Tile[[128, 128], pl.FP32, pl.Mem.Vec] = pl.tile.load(
                     x, [0, 0], [128, 128], [128, 128], target_memory=pl.Mem.Vec
                 )
-                with pl.at(level=pl.Level.CORE_GROUP):
-                    for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.LEFT_RIGHT):
-                        h__tile: pl.Tile[[128, 128], pl.FP32, pl.Mem.Vec] = pl.tile.exp(x__tile)
-                        res__tile: pl.Tile[[128, 256], pl.FP32, pl.Mem.Mat] = pl.tile.aic_gather(h__tile)
+                for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.LEFT_RIGHT):
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_convert_tensor_to_tile_ops.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_lower_auto_vector_split.py [hunk 3]
```diff
@@ -2470,17 +2550,70 @@ def f(
...
             out_0: pl.Out[pl.Tensor[[128, 128], pl.FP32]],
         ) -> pl.Tensor[[128, 128], pl.FP32]:
-            qk = pl.matmul(a_left, b_right)
-            for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.UP_DOWN):  # noqa: B007
-                qk_h = pl.aiv_shard(qk)  # noqa: F841 - explicit boundary
-                t = pl.load(data, [0, 0], [128, 128], target_memory=pl.MemorySpace.Vec)  # full width
-                out_0 = pl.tile.store(t, [0, 0], out_0)
+            with pl.at(level=pl.Level.CORE_GROUP):
+                qk = pl.matmul(a_left, b_right)
+                for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.UP_DOWN):  # noqa: B007
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_lower_auto_vector_split.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_verify_aiv_split.py [hunk 4]
```diff
@@ -980,5 +989,115 @@ def test_crossing_checks_inert_without_a_region():
...
+
+    @pl.program
+    class Prog:
+        @pl.function(type=pl.FunctionType.InCore)
+        def core(
+            self,
+            a: pl.Tensor[[512, 128], pl.FP32],
+            out: pl.Out[pl.Tensor[[512, 128], pl.FP32]],
+        ) -> pl.Tensor[[512, 128], pl.FP32]:
+            for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.NONE):  # noqa: B007
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_verify_aiv_split.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_verify_aiv_split.py [hunk 5]
```diff
@@ -980,5 +989,115 @@ def test_crossing_checks_inert_without_a_region():
...
+    @pl.program
+    class Prog:
+        @pl.function(type=pl.FunctionType.InCore)
+        def core(
+            self,
+            a: pl.Tensor[[512, 128], pl.FP32],
+            out: pl.Out[pl.Tensor[[512, 128], pl.FP32]],
+        ) -> pl.Tensor[[512, 128], pl.FP32]:
+            with pl.at(level=pl.Level.CORE_GROUP):
+                for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.NONE):  # noqa: B007
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_verify_aiv_split.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_verify_aiv_split.py [hunk 6]
```diff
@@ -980,5 +989,115 @@ def test_crossing_checks_inert_without_a_region():
...
+
+    @pl.program
+    class Prog:
+        @pl.function
+        def main(
+            self,
+            a: pl.Tensor[[512, 128], pl.FP32],
+            out: pl.Out[pl.Tensor[[512, 128], pl.FP32]],
+        ) -> pl.Tensor[[512, 128], pl.FP32]:
+            for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.NONE):  # noqa: B007
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_verify_aiv_split.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_verify_aiv_split.py [hunk 7]
```diff
@@ -980,5 +989,115 @@ def test_crossing_checks_inert_without_a_region():
...
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_verify_aiv_split.py]: Fixed in `4ddca96c` — the unused iteration variable is now `_`, matching the existing `for _ in pl.range(...)` idiom already used elsewhere in the test suite. Thanks!

File: tests/ut/ir/transforms/test_lower_auto_vector_split.py [hunk 8]
```diff
@@ -2470,17 +2550,70 @@ def f(
...
+                    qk_h = pl.aiv_shard(qk)  # noqa: F841 - explicit boundary
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_lower_auto_vector_split.py]: Fixed in `4ddca96c` — renamed to `_qk_h`. Note a bare `_` is not usable here: the enclosing region's loop variable is now also `_`, and the DSL parser binds by name, so `_assign_or_let` would resolve the assignment to the loop variable instead of creating a new binding. The underscore prefix still marks it as an intentionally unused explicit-boundary marker.

File: tests/ut/ir/transforms/test_lower_auto_vector_split.py [hunk 9]
```diff
@@ -1966,6 +1966,77 @@ def split_auto(
...
+def test_auto_path_scope_bodied_mixed_is_rejected():
+    """A mixed AUTO ``pl.split`` function behind a scope is rejected."""
+
+    @pl.program
+    class Scoped:
+        @pl.function(type=pl.FunctionType.InCore, attrs={"split": pl.SplitMode.UP_DOWN})
+        def split_auto(
+            qk: pl.Tile[[128, 128], pl.FP32, pl.Mem.Mat],
+            out_0: pl.Out[pl.Tensor[[128, 128], pl.FP32]],
+        ) -> pl.Tensor[[128, 128], pl.FP32]:
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_lower_auto_vector_split.py:1992]: Fixed in `4ddca96c` — added the `self` parameter. The DSL parser skips an unannotated leading `self` (`ast_parser.py:1024`), so the generated IR is unchanged. Thanks!

File: tests/ut/ir/transforms/test_lower_auto_vector_split.py [hunk 10]
```diff
@@ -1966,6 +1966,77 @@ def split_auto(
...
+    silent skip into a spurious hard error.
+    """
+
+    @pl.program
+    class Before:
+        @pl.function(type=pl.FunctionType.InCore, attrs={"split": pl.SplitMode.UP_DOWN})
+        def pure_vec(
+            data: pl.Tensor[[128, 128], pl.FP32],
+            out_0: pl.Out[pl.Tensor[[128, 128], pl.FP32]],
+        ) -> pl.Tensor[[128, 128], pl.FP32]:
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/ut/ir/transforms/test_lower_auto_vector_split.py:2021]: Fixed in `4ddca96c` — added the `self` parameter. The DSL parser skips an unannotated leading `self` (`ast_parser.py:1024`), so the generated IR is unchanged. Thanks!

**Review by lyfne123 — APPROVED** [review]
